From 78289aa6bdca1fe7cd4c7c50c50e15e04328e568 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 28 Apr 2015 22:20:37 +0200
Subject: [PATCH 09/10] qtsystems: add packageconfig for bluez4 and disable it
 by default

* bluez5 is now enabled by default and conflicts with bluez4

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 recipes-qt/qt5/qtsystems.inc | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/recipes-qt/qt5/qtsystems.inc b/recipes-qt/qt5/qtsystems.inc
index 6cdfc8e..3bd4538 100644
--- a/recipes-qt/qt5/qtsystems.inc
+++ b/recipes-qt/qt5/qtsystems.inc
@@ -1,6 +1,15 @@
 require qt5.inc
 
-DEPENDS += "qtbase qtdeclarative udev bluez4 gconf"
+DEPENDS += "qtbase qtdeclarative udev gconf"
+
+PACKAGECONFIG ??= ""
+PACKAGECONFIG[bluez4] = "OE_BLUEZ_ENABLED,,bluez4"
+
+do_configure_prepend() {
+    export QMAKE_CACHE_EVAL="CONFIG+=${EXTRA_OECONF}"
+    # disable bluez test if it isn't enabled by PACKAGECONFIG
+    sed -i 's/^    qtCompileTest(bluez)/    OE_BLUEZ_ENABLED:qtCompileTest(bluez)/g' ${S}/qtsystems.pro
+}
 
 LICENSE = "BSD & LGPL-2.1 & GFDL-1.3"
 # LICENSE files are missing in 5.0.0
-- 
2.4.2

